### PR TITLE
[SQL] `az sql mi failover`: Support failover from primary or secondary point

### PR DESCRIPTION
**Pull request by MilanBrkicFON**
**Description**  

This PR introduce new command on az sql mi module, `failover`. The failover managed instance initiate KillProcess on sql server instance, thus managed instance will failover to another node. 

**Testing Guide**  

`az sql mi failover`
or
`az sql mi failover --replica-type [Primary|ReadableSecondary]`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

File: src/azure-cli/azure/cli/command_modules/sql/commands.py [hunk 1]
```diff
@@ -573,6 +573,7 @@ def load_command_table(self, _):
         g.show_command('show', 'get')
         g.custom_command('list', 'managed_instance_list')
         g.generic_update_command('update', custom_func_name='managed_instance_update', supports_no_wait=True)
+        g.command('failover', 'failover', supports_no_wait=True)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/commands.py:576]: how can we query failover status when add --no-wait?

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 2]
```diff
@@ -4531,3 +4531,75 @@ def test_sql_server_minimal_tls_version(self, resource_group):
...
+                     JMESPathCheck('vCores', v_cores),
+                     JMESPathCheck('storageSizeInGb', storage_size_in_gb),
+                     JMESPathCheck('licenseType', license_type),
+                     JMESPathCheck('sku.tier', edition),
+                     JMESPathCheck('sku.family', family),
+                     JMESPathCheck('sku.capacity', v_cores),
+                     JMESPathCheck('identity', None)]).get_output_in_json()
+
+        # Failover managed instance primary replica
+        self.cmd('sql mi failover -g {resource_group} -n {managed_instance_name}', checks=NoneCheck())
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:4605]: Is there anything to check with failover result?

Comment (MilanBrkicFON) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:4605]: Not really, the response is not holding any information other then - status "Succeeded".

File: src/azure-cli/azure/cli/command_modules/sql/commands.py (hunk 1 above)
Comment (MilanBrkicFON) [reply src/azure-cli/azure/cli/command_modules/sql/commands.py:576]: I guess by sending 
```
GET https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/DejanDuVnetRG/providers/Microsoft.Sql/locations/westus/managedInstanceAzureAsyncOperation/098ac4ab-d57b-40f1-9fb2-d8533df163f9?api-version=2019-06-01-preview
```
, you just need `requestId`.

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/sql/commands.py:576]: Is there a command or SDK method to query failover status?

**Pull request merged**